torchao/prototype/float8nocompile/float8nocompile_linear.py

```diff
@@ -111,7 +113,40 @@ def from_float(
 
 class matmul_with_args_in_hp(torch.autograd.Function):
     @staticmethod
-    def forward(ctx, input_hp, weight_hp, config, linear_mm_config, kernel_algo):
+    def forward(
+        ctx,
+        input_hp: torch.Tensor,
+        weight_hp: torch.Tensor,
+        config: Float8LinearConfig,
+        linear_mm_config: LinearMMConfig,
+        kernel_algo: KernelAlgorithm,
+        no_precompute_for_backward: bool,
+    ):
+        if no_precompute_for_backward:
```

Comment:
nit: just have two top-level functions instead?  People usually expect `torch.autograd.Function` to have a single forward and backward.

Reply:
Done